compliance/virtualmachines/agent/main.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"context"
+	"os"
+	"os/signal"
+
+	"github.com/stackrox/rox/compliance/virtualmachines/agent/cmd"
+	"github.com/stackrox/rox/pkg/logging"
+	"golang.org/x/sys/unix"
+)
+
+var log = logging.LoggerForModule()
+
+func main() {
+	// Create a context that is cancellable on the usual command line signals. Double
+	// signal forcefully exits.
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go func() {
+		sigC := make(chan os.Signal, 1)
+		signal.Notify(sigC, unix.SIGINT, unix.SIGTERM)
+		sig := <-sigC
+		log.Errorf("%s caught, shutting down...", sig)
```

Comment:
Note, that we have now

```go
signal.NotifyContext()
```
that can make code like this a bit cleaner.

Reply:
I tried out `signal.NotifyContext`, but I think I prefer the original version. The issues I saw were:

* No logging of the signal being caught.
* No double press to force exit.

Or is there a way to make ^ work with `signal.NotifyContext`?